cachito/workers/tasks/gomod.py

```diff
@@ -16,45 +17,96 @@
 log = logging.getLogger(__name__)
 
 
+def _find_invalid_gomod_files(bundle_dir, subpaths):
+    """
+    Find all go modules with missing gomod files.
+
+    These files will need to be present in order for the package manager to proceed with
+    fetching the package sources.
+
+    :param RequestBundleDir bundle_dir: the ``RequestBundleDir`` object for the request
+    :param list subpaths: a list of subpaths in the source repository of gomod packages
+    :return: a list containing all non-existing go.mod files across subpaths
+    :rtype: list
+    """
+    invalid_gomod_files = []
+    for subpath in subpaths:
+        bundle_dir_subpath = bundle_dir.app_subpath(subpath)
+        package_gomod_rel_path = bundle_dir_subpath.relpath(bundle_dir_subpath.go_mod_file)
+        log.debug("Testing for go mod file in {}".format(package_gomod_rel_path))
+        if not bundle_dir_subpath.go_mod_file.exists():
+            invalid_gomod_files.append(package_gomod_rel_path)
+
+    return invalid_gomod_files
+
+
 @app.task
-def fetch_gomod_source(request_id, dep_replacements=None):
+def fetch_gomod_source(request_id, dep_replacements=None, package_configs=None):
     """
     Resolve and fetch gomod dependencies for a given request.
 
     :param int request_id: the Cachito request ID this is for
     :param list dep_replacements: dependency replacements with the keys "name" and "version"
+    :param list package_configs: the list of optional package configurations submitted by the user
```

Comment:
Dependency replacements with multiple packages is a tricky topic. I don't think there is currently any consensus on whether we want to apply the replacements globally or per-package (or allow both?).

Maybe a reasonable thing to do would be to fail the request if there is more than one package and dependency replacements are defined (until we settle on an approach)? CC @athos-ribeiro @lcarva

Reply:
I am fine with that if you want to go that approach...just let me know 😄

Reply:
+1 for throwing an error as suggested.

Reply:
We now issue an error if we try to do a dependency replacement with more than one path specified.